### PR TITLE
first pass at adding wallet interface and soroban signing

### PR DESCRIPTION
**Pull request by piyalbasu**

### CODE REVIEW

File: src/walletSdk/Wallet/index.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import FreighterApi from "@stellar/freighter-api";
```
Comment (willemneal) [inline src/walletSdk/Wallet/index.ts:1]: Perhaps it's best if we flipped this. This way the frieghter API package could provide the Wallet implementation.

Comment (piyalbasu) [reply src/walletSdk/Wallet/index.ts:1]: Ah yeah, that makes sense. The idea being that the freighter-api is the default interface that we expect from a wallet.

For added context: We're also talking about creating some helper methods in this lib to allow other browser extensions to quickly implement this expected interface in their own codebase so they can quickly make use of `signAndSend` and `signAuthEntries`

File: package.json [hunk 2]
```diff
@@ -41,7 +41,8 @@
     "webpack-cli": "^5.1.1"
   },
   "dependencies": {
-    "@stellar/stellar-sdk": "^11.1.0",
+    "@stellar/freighter-api": "^1.7.1",
+    "@stellar/stellar-sdk": "stellar/js-stellar-sdk#bindings",
```
Comment (CassioMG) [inline package.json:45]: @piyalbasu just out of curiosity, what's the reason behind this "#bindings" change?

Comment (piyalbasu) [reply package.json:45]: This was targeted at a branch of stellar-sdk that had the TS bindings update for dev purposes. This work may be merged already. I'll follow up

**Pull request closed without merging**